### PR TITLE
Pull in HMAC component for Mnemonic (BIP39) support

### PR DESCRIPTION
**Pull request by sinetek**
This pulls in pkcs5_pbkdf2_hmacsha512 which is an implementation of PKCS5 PBKDF2 using the HMAC-SHA512 function on sizes 64. Required for implementing BIP39 in CMnemonic.
Comes along with test vectors to prove.

### CODE REVIEW

**Comment by aguycalled** [comment]
tested locally, tests pass, verified test vectors matches those of other implementations

**Review by aguycalled — APPROVED** [review]

**Comment by mxaddict** [comment]
Test on Ubuntu 20.04 tests pass on x86_64

But on ci when building for 32bit build is failing with this error:
```
crypto/pkcs5_pbkdf2.cpp:67:29: error: ‘bit_xor’ is not a member of ‘std’
```

which makes no sense to me as I'm sure bit_xor is part of std since std++17 no?

**Comment by sinetek** [comment]
> Test on Ubuntu 20.04 tests pass on x86_64
> 
> But on ci when building for 32bit build is failing with this error:
> 
> ```
> crypto/pkcs5_pbkdf2.cpp:67:29: error: ‘bit_xor’ is not a member of ‘std’
> ```
> 
> which makes no sense to me as I'm sure bit_xor is part of std since std++14 no?

oh aparently it is in Defined in header <functional>
sorry about that need to include the right file😅

**Review by mxaddict — DISMISSED** [review]

File: src/crypto/pkcs5_pbkdf2.h [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// Copyright (c) 2022 The Navcoin developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#pragma once
```
Comment (mxaddict) [inline src/crypto/pkcs5_pbkdf2.h:5]: I think this might need to be changed to ifndef instead of pragma once since linter is complaining about it.

Comment (mxaddict) [reply src/crypto/pkcs5_pbkdf2.h:5]: Like so:
```cpp
#ifndef BITCOIN_CRYPTO_PKCS5_PBKDF2_H
#define BITCOIN_CRYPTO_PKCS5_PBKDF2_H
```

File: src/crypto/pkcs5_pbkdf2.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        CHMAC_SHA512 hmac (P.data(), P.size());   // key
+        hmac.Write(digest1.data(), digest1.size());    // data
+        hmac.Finalize(digest2.data());
+
+        digest1.assign(digest2.begin(), digest2.end());
+
+        std::transform(buffer.begin(), buffer.end(),
+                       digest1.begin(),
+                       buffer.begin(),
+                       std::bit_xor());
```
Comment (mxaddict) [inline src/crypto/pkcs5_pbkdf2.cpp:68]: Should be updated to use `std::bit_xor<uint8_t>()` as discussed in discord

**Comment by sinetek** [comment]
Updated. Changed the include guard to standard one, included type for std::bit_xor.

**Review by mxaddict — APPROVED** [review]

**Pull request merged**